kwok/kwok_script_runner.go

```diff
@@ -0,0 +1,86 @@
+package kwok
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+)
+
+// GetKubeconfig get the full path to the kwok directory's
+// kubeconfig file, no matter where the functions is called from.
+func getKubeconfig() (string, error) {
+	_, filename, _, ok := runtime.Caller(0)
+	if !ok {
+		return "", fmt.Errorf("failed to find the path of the current script")
+	}
+	dirPath := filepath.Dir(filename)
+	kubeconfigPath := filepath.Join(dirPath, "kubeconfig")
+	return kubeconfigPath, nil
+}
+
+// setKubeconfig set the kubeconfig using an environment variable.
+// It eithers get the full path to the kubeconfig, or set it to the
+// one in the kwok directory.
+func setKubeconfig(kubeconfigPath string) error {
+	if kubeconfigPath == "" {
+		kubeconfigPath, _ = getKubeconfig()
+	}
+	os.Setenv("KUBECONFIG", kubeconfigPath)
+
+	return nil
+}
+
+type ScriptExecutor interface {
+	Execute(scriptPath string) ([]byte, error)
+}
+
+type ShellScriptExecutor struct{}
+
+func (s *ShellScriptExecutor) Execute(scriptPath string) ([]byte, error) {
+	return exec.Command(scriptPath).CombinedOutput()
+}
+
+func ExecuteScript(executor ScriptExecutor, scriptPath string) ([]byte, error) {
+	output, err := executor.Execute(scriptPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to execute command\nOutput: %s\nError: %w", string(output), err)
+	}
+	return output, nil
+}
+
+func ParseScript(output []byte) (map[string]interface{}, error) {
```

Comment:
This is basically an opaque abstraction over `json.Unmarshal` better drop it.

Also the name does not do a good job of indicating its merely parsing JSON.

If you still want to keep it to make the tests simpler, then make it more effective by turning it into an assert-like function that handles the error and only returns in the case the output is right.